src/Components/test/testassets/BasicTestApp/EventCallbackTest/EventCallbackCases.cshtml

```diff
@@ -0,0 +1,36 @@
+@*
+    Test cases for using EventCallback with various delegate scenarios that used to be troublesome.
+
+    Currently these cases are **VERBOSE** because we haven't yet landed the compiler support for EventCallback.
+    This will be cleaned up soon, and all of the explicit calls to EventCallback.Factory will go away.
+*@
+<div>
+    <p>Clicking any of these buttons should cause the count to go up by one!</p>
+    <p id="render_count">Render Count: @(++renderCount)</p>
+</div>
+<div id="capturing_lambda">
+    <h3>Passing Capturing Lambda to Button</h3>
+    <InnerButton OnClick="@EventCallback.Factory.Create(this, () => { GC.KeepAlive(this); })" Text="Capturing Lambda" />
+</div>
+<div id="unbound_lambda">
+    <h3>Passing Unbound Lambda to Button</h3>
+    <InnerButton OnClick="@EventCallback.Factory.Create(this, () => { })" Text="Unbound Lambda" />
+</div>
+<div id="unbound_lambda_nested">
+    <h3>Passing Unbound Lambda to Nested Button</h3>
+    <MiddleButton OnClick="@EventCallback.Factory.Create(this, () => { })" Text="Unbound Lambda Nested" />
+</div>
+<div id="unbound_lambda_strongly_typed">
+    <h3>Passing Capturing Lambda to Strongly Typed Button</h3>
+    <StronglyTypedButton OnClick="@(EventCallback.Factory.Create<UIMouseEventArgs>(this, () => { GC.KeepAlive(this); }))" Text="Unbound Lambda Strongly-Typed" />
+</div>
+<div id="unbound_lambda_child_content">
+    <h3>Passing Child Content</h3>
+    <TemplatedControl>
+        <button onclick="@EventCallback.Factory.Create(this, () => { })">Unbound Lambda Child Content</button>
+    </TemplatedControl>
+</div>
```

Comment:
@SteveSandersonMS anything else you think is worth adding?

Reply:
When you do `bind-Value` to a component, does the `ValueChanged` invocation now implicitly trigger `StateHasChanged` on the outer component? Not sure whether that is meant to happen automatically given the codegen changes, but if so, would be good to show that too.

Reply:
> When you do `bind-Value` to a component, does the `ValueChanged` invocation now implicitly trigger `StateHasChanged` on the outer component?

Using `Action<T>` will always trigger an event on the delegate target (if possible). We'd recommend using `EventCallback<T>` for this case, which will be supported. 

I'll add a test case for this

Reply:
Note that this requires the component that is producing the bindable to use `EventCallback<T> ValueChanged`.